packages/rum/src/domain/record/startFullSnapshots.spec.ts

```diff
@@ -29,26 +34,68 @@ describe('startFullSnapshots', () => {
     )
   })
 
+  afterEach(() => {
+    window.requestIdleCallback = originalRequestIdleCallback
+    window.cancelIdleCallback = originalCancelIdleCallback
+  })
+
   it('takes a full snapshot when startFullSnapshots is called', () => {
     expect(fullSnapshotCallback).toHaveBeenCalledTimes(1)
   })
 
   it('takes a full snapshot when the view changes', () => {
+    const { triggerIdleCallbacks } = mockRequestIdleCallback()
+
     lifeCycle.notify(LifeCycleEventType.VIEW_CREATED, {
       startClocks: viewStartClock,
     } as Partial<ViewCreatedEvent> as any)
 
+    triggerIdleCallbacks()
+
     expect(fullSnapshotCallback).toHaveBeenCalledTimes(2)
   })
 
+  it('cancels the previous idle callback when the view changes', () => {
+    const { triggerIdleCallbacks, cancelIdleCallbackSpy } = mockRequestIdleCallback()
+
+    lifeCycle.notify(LifeCycleEventType.VIEW_CREATED, {
+      startClocks: viewStartClock,
+    } as Partial<ViewCreatedEvent> as any)
+
+    lifeCycle.notify(LifeCycleEventType.VIEW_CREATED, {
+      startClocks: viewStartClock,
+    } as Partial<ViewCreatedEvent> as any)
+
+    triggerIdleCallbacks()
+    expect(cancelIdleCallbackSpy).toHaveBeenCalledTimes(1)
+  })
```

Comment:
**💬 suggestion:** ‏Ideally, tests should be assertions on the expected behavior of the subject, not how it's implemented. Instead of testing whether the idle callback has been canceled (which is an implementation detail), test that we end up having a single FS

```suggestion
  it('cancels the full snapshot if another view is created before it can it happens', () => {
    const { triggerIdleCallbacks, cancelIdleCallbackSpy } = mockRequestIdleCallback()

    lifeCycle.notify(LifeCycleEventType.VIEW_CREATED, {
      startClocks: viewStartClock,
    } as Partial<ViewCreatedEvent> as any)

    lifeCycle.notify(LifeCycleEventType.VIEW_CREATED, {
      startClocks: viewStartClock,
    } as Partial<ViewCreatedEvent> as any)

    triggerIdleCallbacks()
    expect(fullSnapshotCallback).toHaveBeenCalledTimes(2)
  })
```